activerecord/test/cases/relations_test.rb

```diff
@@ -119,6 +119,17 @@ def test_loaded_first
     assert topics.loaded?
   end
 
+  def test_loaded_first_with_limit
+    topics = Topic.all.order('id ASC')
+
+    assert_queries(1) do
+      topics.to_a # force load
```

Comment:
You could move this outside the `assert_queries` block. Then we don't have to do the actual code 2 times and we can assert that no queries are executed.